lib/manageiq/providers/openstack/legacy/openstack_handle/handle.rb

```diff
@@ -352,7 +352,7 @@ def tenant_accessible?(name)
     end
 
     def default_tenant_name
-      return @default_tenant_name ||= "admin" if tenant_accessible?("admin")
+      return @default_tenant_name ||= ("admin" if tenant_accessible?("admin"))
```

Comment:
If this `tenant_accessible?("admin")` returned false then the memoized `@default_tenant_name` was never being returned

Reply:
is the code supposed to be doing the return only `if tenant_accessible?("admin")`

Suggest majorly simplifying as follows (or similar to make the logic clearer)
```ruby
if tenant_accessible?("admin")
  @default_tenant_name ||= "admin"
  return @default_tenant_name
end
```

Reply:
looking at the whole method, maybe even the following would be better:

```ruby
def default_tenant_name
  @default_tenant_name ||= determine_tenant_name
end

def determine_tenant_name
  return "admin" if tenant_accessible?("admin")
  
  tenant_names.each do |name|
    next if name == "services"
    return name if tenant_accessible?(name)
  end

  nil
end
```

Reply:
The code is supposed to be returning `@default_tenant_name` if it is set, otherwise detecting what the tenant_name is (with "admin" as the default).

:+1: splitting this logic up into two methods will be more clear